packages/apps/dms/src/api/document-api.js

```diff
@@ -145,3 +187,75 @@ export async function searchDirectory({searchValue, authorId, page = 0}) {
     provider: 'model',
   });
 }
+
+export async function createDocument({document, model, modelId}) {
+  const {matchers} = formatRequestBody(document, 'data');
+
+  return getActionApi().send({
+    url: '/ws/rest/com.axelor.dms.db.DMSFile',
+    method: 'put',
+    body: {
+      data: {
+        relatedModel: model,
+        relatedId: modelId,
+        ...document,
+      },
+    },
+    description: 'create document',
+    matchers: {
+      modelName: 'com.axelor.dms.db.DMSFile',
+      id: Date.now(),
+      fields: matchers,
+    },
+  });
+}
+
+export async function updateDocument({document}) {
+  const {matchers} = formatRequestBody(document, 'data');
+
+  return getActionApi().send({
+    url: '/ws/rest/com.axelor.dms.db.DMSFile',
+    method: 'post',
+    body: {
+      data: document,
+    },
+    description: 'update document',
+    matchers: {
+      modelName: 'com.axelor.dms.db.DMSFile',
+      id: document.id,
+      fields: {
+        ...matchers,
+      },
+    },
+  });
+}
+
+export async function fetchDirectory({model, modelId}) {
+  return createStandardSearch({
+    model: 'com.axelor.dms.db.DMSFile',
+    criteria: createFetchDirectoryCriteria({model, modelId}),
+    fieldKey: 'dms_document',
+    sortKey: 'dms_document',
+    page: 0,
+    numberElementsByPage: 1,
+    provider: 'model',
+  });
+}
+
+export async function countAttachedFiles({model, modelId}) {
+  return axiosApiProvider.post({
```

Comment:
Use createStandardSearch here and transform domain into criterias please so it can be used offline too